apps/language_server/lib/language_server/providers/folding_range/special_token.ex

```diff
@@ -55,23 +55,46 @@ defmodule ElixirLS.LanguageServer.Providers.FoldingRange.SpecialToken do
 
   @spec group_tokens([Token.t()]) :: [[Token.t()]]
   defp group_tokens(tokens) do
-    tokens
-    |> Enum.reduce([], fn
-      {:identifier, _, identifier} = token, acc when identifier in [:doc, :moduledoc] ->
-        [[token] | acc]
+    do_group_tokens(tokens, [])
+  end
 
-      {k, _, _} = token, [[{:identifier, _, _}] = head | tail] when k in @kinds ->
-        [[token | head] | tail]
+  defp do_group_tokens([], acc), do: acc
 
-      {k, _, _} = token, acc when k in @kinds ->
-        [[token] | acc]
+  # Don't create folding ranges from `@doc false` and `@moduledoc false`
+  defp do_group_tokens([{:identifier, _, doc_identifier}, {false, _, _} | rest], acc)
+       when doc_identifier in [:doc, :moduledoc] do
+    do_group_tokens(rest, acc)
+  end
 
-      {:eol, _, _} = token, [[{k, _, _} | _] = head | tail] when k in @kinds ->
-        [[token | head] | tail]
+  # Start a folding range for `@doc` and `@moduledoc`
+  defp do_group_tokens([{:identifier, _, doc_identifier} = token | rest], acc)
+       when doc_identifier in [:doc, :moduledoc] do
```

Comment:
typedoc?

Reply:
Yeah, that's probably a good idea. Handled in acb5255e26e67ad46c26a1343f123e9acc49fc85